js/style/style_constant.js

```diff
@@ -2,26 +2,38 @@
 
 var util = require('../util/util');
 
-exports.resolve = function(value, constants) {
-    function resolve(value) {
-        return typeof value === 'string' && value[0] === '@' ? constants[value] : value;
+/**
+ * Given a value that may be a constant reference and an object
+ * of constants, try to detect whether the value is a constant reference
+ * and return the constant value if so. This will also try to resolve
+ * any constants used in composite values like ramp stops.
+ *
+ * @param {*} value any value in a layout or paint property
+ * @param {Object} constants object with constant values
+ * @returns {*} a resolved value
+ * @private
+ */
+exports.resolve = function resolve(value, constants) {
+    function resolveInner(value) {
+        return typeof value === 'string' && value[0] === '@' ? constants[value].value : value;
     }
```

Comment:
can we avoid creating a closure on every call by putting the function outside with a `constaints` second argument?